src/commonMain/kotlin/fr/acinq/lightning/wire/OfferTypes.kt

```diff
@@ -0,0 +1,1003 @@
+                LightningCodecs.writeU16(address.value.size(), out)
+                LightningCodecs.writeBytes(address.value, out)
+            }
+        }
+
+        companion object : TlvValueReader<InvoiceFallbacks> {
+            const val tag: Long = 172
+            override fun read(input: Input): InvoiceFallbacks {
+                val addresses = ArrayList<FallbackAddress>()
+                while (input.availableBytes > 0) {
+                    val version = LightningCodecs.byte(input)
+                    val value = ByteVector(LightningCodecs.bytes(input, LightningCodecs.u16(input)))
+                    addresses.add(FallbackAddress(version, value))
+                }
+                return InvoiceFallbacks(addresses)
+            }
+        }
+    }
+
+    /**
+     * Features supported to pay the invoice.
+     */
+    data class InvoiceFeatures(val features: Features) : InvoiceTlv {
+        override val tag: Long get() = InvoiceFeatures.tag
+
+        override fun write(out: Output) {
+            LightningCodecs.writeBytes(features.toByteArray(), out)
+        }
+
+        companion object : TlvValueReader<InvoiceFeatures> {
+            const val tag: Long = 174
+            override fun read(input: Input): InvoiceFeatures {
+                return InvoiceFeatures(Features(LightningCodecs.bytes(input, input.availableBytes)))
+            }
+        }
+    }
+
+    /**
+     * Public key of the invoice recipient.
+     */
+    data class InvoiceNodeId(val nodeId: PublicKey) : InvoiceTlv {
+        override val tag: Long get() = InvoiceNodeId.tag
+
+        override fun write(out: Output) {
+            LightningCodecs.writeBytes(nodeId.value, out)
+        }
+
+        companion object : TlvValueReader<InvoiceNodeId> {
+            const val tag: Long = 176
+            override fun read(input: Input): InvoiceNodeId {
+                return InvoiceNodeId(PublicKey(LightningCodecs.bytes(input, input.availableBytes)))
+            }
+        }
+    }
+
+    /**
+     * Signature from the sender when used in an invoice request.
+     * Signature from the recipient when used in an invoice.
+     */
+    data class Signature(val signature: ByteVector64) : InvoiceRequestTlv, InvoiceTlv {
+        override val tag: Long get() = Signature.tag
+
+        override fun write(out: Output) {
+            LightningCodecs.writeBytes(signature, out)
+        }
+
+        companion object : TlvValueReader<Signature> {
+            const val tag: Long = 240
+            override fun read(input: Input): Signature {
+                return Signature(ByteVector64(LightningCodecs.bytes(input, input.availableBytes)))
+            }
+        }
+    }
+
+    fun filterOfferFields(tlvs: TlvStream<InvoiceRequestTlv>): TlvStream<OfferTlv> {
+        // Offer TLVs are in the range (0, 80).
+        return TlvStream<OfferTlv>(
+            tlvs.records.filterIsInstance<OfferTlv>().toSet(),
+            tlvs.unknown.filter{it.tag < 80}.toSet()
```

Comment:
This matches what we do in eclair, but I don't really remember the spec, can you refresh my memory? That means we don't use the "standard" mechanisms for unknown TLVs here (a TLV with tag 65501 would be dropped), is that expected?